types/src/share.rs

```diff
@@ -33,36 +33,37 @@ pub struct NamespacedRow {
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 #[serde(try_from = "RawShare", into = "RawShare")]
 pub struct Share {
-    pub namespace: Namespace,
-    pub data: Vec<u8>,
+    pub data: [u8; appconsts::SHARE_SIZE],
 }
 
 impl Share {
-    pub fn new(mut bytes: Vec<u8>) -> Result<Self> {
-        if bytes.len() != appconsts::SHARE_SIZE {
-            return Err(Error::InvalidShareSize(bytes.len()));
+    pub fn new(data: &[u8]) -> Result<Self> {
+        if data.len() != appconsts::SHARE_SIZE {
+            return Err(Error::InvalidShareSize(data.len()));
         }
-
-        let namespace = Namespace::from_raw(&bytes[..NS_SIZE])?;
-        bytes.drain(..NS_SIZE);
+        Namespace::from_raw(&data[..NS_SIZE])?;
```

Comment:
Add a comment here that we do it for validation because later we will use the unchecked version to create the namespace.

Reply:
done